File: Linked Lists/BFS_graphs.java
import java.util.*;

public class BFS_graphs {
  public ArrayList<Integer> BfsTraversal(int v, ArrayList<ArrayList<Integer>> adj) {
    boolean[] visited = new boolean[v];
    ArrayList<Integer> bfs = new ArrayList<>();
    Queue<Integer> q = new LinkedList<>();

    q.add(0);
    visited[0] = true;

    while (!q.isEmpty()) {
      int node = q.poll();
      bfs.add(node);

      for (Integer it : adj.get(node)) {
        if (!visited[it]) {
          visited[it] = true;
          q.add(it);
        }
      }
    }
    return bfs;
  }
}

File: Linked Lists/BFS_graphs.cpp
#include <stdio.h>
#include <stdlib.h>
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

class BFS_graphs
{

public:
  // Function to return breadth first traversal items of a graph
  // V is the number of nodes, adj is the given adjacency list
  vector<int> BfsTraversal(int V, vector<int> adj[]) {
    vector<int> bfs;
    int visited[V] = {0};
    queue<int> q;
    q.push(0);
    visited[0] = 1;

    while (!q.empty()) {
      int node = q.front();
      q.pop();
      bfs.push_back(node);
      for (auto item : adj[node]) {
        if (! visited[item]) {
          visited[item] = 1;
          q.push(item);
        }
      }
    }
    return bfs;
  }
};

